components/ResizeableWindow.jsx

```diff
@@ -102,10 +102,18 @@ class ResizeableWindow extends React.Component {
         }
     }
     computeInitialX = (x) => {
-        return x >= 0 ? x : window.innerWidth - Math.abs(x);
+        if (this.props.dockable == "right") {
+            return x > 0 ? x : window.innerWidth - this.props.initialWidth - Math.abs(x);
+        } else {
+            return x >= 0 ? x : window.innerWidth - this.props.initialWidth - Math.abs(x);
+        }
```

Comment:
@danceb Reading this change more carefully, I cannot understand why it is needed? InitialX and initialY are actually only used in the floating state, so modifiying the behaviour based on the dockable state looks wrong. Also, say with `dockable=right`and `initialX=0`, having the left window border at the right screen border is most likely not what you want, since it would just mean that the window is out of sight. Can you elaborate the rationale for this?

Reply:
I thought, that it is more obvious, if the docking side is the one where the initial values are related to. So if you specify `dockable=right` and `initialX=0` without that change, it would place the floating window on the left hand side. Which is uncommon in my opinion as clicking the docking button will then dock the window to the opposite side and furthermore there is no chance to place the window on the right hand side.
With `return x > 0 ? x : window.innerWidth - this.props.initialWidth - Math.abs(x);` you will have the right window border at the right screen border, not the left one, as the `initialWidth` is also considered here. Calculating the position without `initialWidth` as it was before led to a placement out of sight. 
Does this make sense to you?

Reply:
Right, I missed the `- this.props.initialWidth` part. I'd still prefer to not mix the floating positioning with the docking state, an elegant solution IMO would be to treat negative 0 as right and positive zero as left (resp bottom/top). Indeed it is possible to distinguish between positive and negative zero via `Object.is`. What do you think?

Reply:
Ah perfect! Didn't know about this option. So using `Object.is` will lead to this:
```javascript
    computeInitialX = (x) => {
        return x > 0 || Object.is(x, 0) ? x : window.innerWidth - this.props.initialWidth - Math.abs(x);
    };
    computeInitialY = (y) => {
        return y > 0 || Object.is(y, 0) ? y : window.innerHeight - this.props.initialHeight - Math.abs(y) - 100;
    };
```